v2/robotmk/scheduler.py

```diff
@@ -1,96 +1,151 @@
 """Scheduler"""
 
 import argparse
+import dataclasses
 import datetime
 import pathlib
 import subprocess
-from collections.abc import Iterable, Mapping, Sequence
-from typing import Final
+from collections.abc import Iterable, Iterator, Mapping, Sequence
+from typing import Final, assert_never
 from uuid import uuid4
 
 from apscheduler.schedulers.blocking import BlockingScheduler  # type: ignore[import]
 from apscheduler.triggers.interval import IntervalTrigger  # type: ignore[import]
-from pydantic import BaseModel
+from pydantic import BaseModel, TypeAdapter
 from robot import rebot  # type: ignore[import]
 
 from robotmk.environment import RCCEnvironment, ResultCode, RobotEnvironment
 from robotmk.runner import Attempt, RetrySpec, RetryStrategy, Variant, create_attempts
 from robotmk.session import CurrentSession, UserSession
 
 
-class _RCC(BaseModel, frozen=True):
-    binary: pathlib.Path
-    robot_yaml: pathlib.Path
+class _RCCConfig(BaseModel, frozen=True):
+    binary_path: pathlib.Path
 
 
-class _UserSession(BaseModel, frozen=True):
+class _UserSessionConfig(BaseModel, frozen=True):
     user_name: str
 
 
-class _SuiteConfig(BaseModel, frozen=True):  # pylint: disable=too-few-public-methods
+class _SuiteConfig(BaseModel, frozen=True):
     execution_interval_seconds: int
     robot_target: pathlib.Path
-    working_directory: pathlib.Path
     variants: Sequence[Variant]
     retry_strategy: RetryStrategy
-    env: _RCC | None
-    session: _UserSession | None
+    session: _UserSessionConfig | None
+
+
+class _SystemPythonSuiteConfig(_SuiteConfig, frozen=True):
+    ...
+
 
+class _RCCSuiteConfig(_SuiteConfig, frozen=True):
+    robot_yaml_path: pathlib.Path
 
-class _SchedulerConfig(BaseModel, frozen=True):
-    suites: Mapping[str, _SuiteConfig]
 
+class _ConfigSystemPython(BaseModel, frozen=True):
+    rcc: None
```

Comment:
In the original version, it couldn't be removed, otherwise, any config would deserialize to the system python case. However, I made this more explicit now anyway.